lib/restbase.js

```diff
@@ -143,6 +143,9 @@ RESTBase.prototype.request = function (req) {
 RESTBase.prototype._request = function (req) {
     var self = this;
 
+    req.headers = req.headers || {};
+    req.headers['x-request-id'] = req.headers['x-request-id'] || rbUtil.generateRequestId();
```

Comment:
We leave them as-is when we [clone the request](https://github.com/wikimedia/restbase/blob/master/lib/restbase.js#L22). That's what this PR is based on :)